ldoc/html.lua

```diff
@@ -323,6 +323,17 @@ function ldoc.source_ref (fun)
 
    check_directory(args.dir) -- make sure output directory is ok
 
+   -- project icon
+   if ldoc.icon then
+      local dir_data = args.dir .. '/data'
+      if not path.isdir(dir_data) then
+          local lfs = require 'lfs'
```

Comment:
I don't think this addition is correct actually, `lfs` in included in the namespace already via the very top level `ldoc.lua`.

Reply:
It probably isn't necessary. But when left out, the luacheck reports an error/warning about unrecognized "lfs". Should I remove it anyway?

Reply:
Luacheck doesn't always understand the scope, especially when something in a different file is accessible in the namespace. I suggest adding a lint exception comment to avoid the lint error but don't duplicate the require unless we prove its necessary. The way I would test it is to first remove the data directory, then change the line to something bogus like `local lfs = require 'foobar'` and make sure you get an error. That will prove you are actually running the code in question. Then try removing the line entirely. If it doesn't error saying `lfs` is nil then you're good without the import.

Reply:
It's as you say, works without any error/warning with the call to `require` removed.

> I suggest adding a lint exception

I'm not sure how to do that. Is that done in `.luacheckrc`?